### PR TITLE
W-10781615 - SDK forward compatibility for Object Store - Support injection of the SDK API Object Store

### PR DESCRIPTION
**Pull request by dabreu**

### CODE REVIEW

File: integration/pom.xml [hunk 1]
```diff
@@ -57,6 +57,12 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.mule.sdk</groupId>
+            <artifactId>mule-sdk-api</artifactId>
+            <version>1.0.0-SNAPSHOT</version>
```
Comment (marianogonzalez) [inline integration/pom.xml:63]: is the version necessary? don't we have dependency management on this? If DM not available here, then at least use a property

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: integration/src/test/java/org/mule/test/components/ExtensionWithObjectStoreTestCase.java [hunk 2]
```diff
@@ -28,6 +29,9 @@ public class ExtensionWithObjectStoreTestCase extends AbstractIntegrationTestCas
   @Inject
   private ObjectStoreManager objectStoreManager;
 
+  @Inject
+  private org.mule.sdk.api.store.ObjectStoreManager sdkObjectStoreManager;
```
Comment (ndinu) [inline integration/src/test/java/org/mule/test/components/ExtensionWithObjectStoreTestCase.java:33]: For the case of the test done, the connector that injects the SDK-API ObjectStoreManager uses the same sdk-api version as the runtime. Would the injection work if the connector uses a newer or older version of the SDK-API?

Comment (dabreu) [reply integration/src/test/java/org/mule/test/components/ExtensionWithObjectStoreTestCase.java:33]: Not sure if I get your point...the injection would work on both runtime and sdk-api 4.5+

File: integration/pom.xml (hunk 1 above)
Comment (dabreu) [reply integration/pom.xml:63]: not necessary, my bad, I will remove it

**Review by marianogonzalez — APPROVED** [review]

**Pull request merged**